src/main/java/com/jwplayer/southpaw/state/RocksDBState.java

```diff
@@ -304,11 +295,23 @@ public void close() {
         writeOptions = null;
         rocksDB = null;
 
+        closeBackupEngine();
+
         if (s3Helper != null) {
             s3Helper.close();
+            s3Helper = null;
         }
     }
 
+    private void closeBackupEngine() {
+        logger.info("Closing RocksDB backup engine");
+        backupOptions.close();
+        backupEngine.close();
+
+        backupOptions = null;
```

Comment:
if we explicitly set this to null do we need to guard against a NPE for the close operations? Also, do the close operations except?

Reply:
done